src/checker/dd/DDAlternatingChecker.cpp

```diff
@@ -93,39 +93,30 @@ void DDAlternatingChecker::postprocess() {
   if (isDone()) {
     return;
   }
-
-  // sum up the contributions of garbage qubits
-  taskManager1.reduceGarbage(functionality);
-  if (isDone()) {
-    return;
-  }
-  taskManager2.reduceGarbage(functionality);
-  if (isDone()) {
-    return;
-  }
 }
 
 EquivalenceCriterion DDAlternatingChecker::checkEquivalence() {
-  // create the full identity matrix
-  auto goalMatrix = dd->makeIdent(nqubits);
-  dd->incRef(goalMatrix);
-
-  // account for any garbage
-  taskManager1.reduceGarbage(goalMatrix);
-  taskManager2.reduceGarbage(goalMatrix);
-
-  taskManager1.reduceAncillae(goalMatrix);
-  taskManager2.reduceAncillae(goalMatrix);
+  std::vector<bool> garbage(nqubits);
+  // is it guaranteed that the two circuits have the same garbage qubits?
+  for (qc::Qubit q = 0U; q < nqubits; ++q) {
+    garbage[static_cast<std::size_t>(q)] =
+        qc1.logicalQubitIsGarbage(q) && qc2.logicalQubitIsGarbage(q);
+  }
```

Comment:
```suggestion
  for (qc::Qubit q = 0U; q < nqubits; ++q) {
    garbage[static_cast<std::size_t>(q)] =
        qc1.logicalQubitIsGarbage(q) && qc2.logicalQubitIsGarbage(q);
  }
```
I am fairly sure this has to be the case. Otherwise both circuits might not have the same number of measured qubits. I still like the idea of setting this explicitly here. At least until we perform stricter checking on the circuits themselves and whether they are "compatible".